src/java/voldemort/store/metadata/MetadataStore.java

```diff
@@ -531,31 +531,34 @@ public Object getCapability(StoreCapabilityType capability) {
             String key = ByteUtils.getString(keyBytes.get(), "UTF-8");
 
             if(METADATA_KEYS.contains(key) || this.storeNames.contains(key)) {
-                List<Versioned<byte[]>> values = Lists.newArrayList();
+                try {
+                    List<Versioned<byte[]>> values = Lists.newArrayList();
 
-                // Get the cached value and convert to string
-                Versioned<String> value = convertObjectToString(key, metadataCache.get(key));
+                    // Get the cached value and convert to string
+                    Versioned<String> value = convertObjectToString(key, metadataCache.get(key));
 
-                // Metadata debugging information
-                if(logger.isTraceEnabled())
-                    logger.trace("Key " + key + " requested, returning: " + value.getValue());
+                    // Metadata debugging information
+                    if(logger.isTraceEnabled())
+                        logger.trace("Key " + key + " requested, returning: " + value.getValue());
 
-                values.add(new Versioned<byte[]>(ByteUtils.getBytes(value.getValue(), "UTF-8"),
-                                                 value.getVersion()));
+                    values.add(new Versioned<byte[]>(ByteUtils.getBytes(value.getValue(), "UTF-8"),
+                                                     value.getVersion()));
 
-                return values;
+                    return values;
+                } catch(Exception e) {
+                    throw new VoldemortException("Failed to read metadata key:"
+                                                         + ByteUtils.getString(keyBytes.get(),
+                                                                               "UTF-8")
+                                                         + " delete config/.temp config/.version directories and restart.",
+                                                 e);
+                }
             } else {
-                throw new VoldemortException("Unhandled Key:" + key + " for MetadataStore get()");
+                throw new VoldemortException("Store " + key + " does not exist on node "
```

Comment:
You are right, the code could be exercised in 2 paths.

1) client Bootstrap, when it tries to do the getStoreClient call.
2) during vadmin.sh meta get <key> 

the second one will throw an confusing exception, but only SRE should run it and first one is more important for our partners to understand.